tools/virsorter/virsorter_run.xml

```diff
@@ -0,0 +1,235 @@
+<tool id="virsorter" name="VirSorter" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0" license="MIT">
+    <description>identify DNA and RNA virus</description>
+    <macros>
+        <token name="@TOOL_VERSION@">2.2.4</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">virsorter</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">virsorter</requirement>
+        <!-- virsorter (virsorter setup -d db -j 4) creates a conda env yaml (and conda env)
+            that is used by the snakemake pipeline. in the tool we disable the use of the conda env. 
+            
+            the following are the pinned requirements used in the conda env that would be created by the setup 
+          -->
+        <requirement type="package" version="8.1.7">click</requirement>
+        <requirement type="package" version="1548">last</requirement>
+        <requirement type="package" version="0.3.3">ncbi-genome-download</requirement>
+        <requirement type="package" version="0.16.12">ruamel.yaml</requirement>
+        <requirement type="package" version="2.6.3">prodigal</requirement>
+        <requirement type="package" version="3.8.19">python</requirement>
+        <requirement type="package" version="1.1.3">screed</requirement>
+        <requirement type="package" version="3.4">hmmer</requirement>
+        <requirement type="package" version="0.22.1">scikit-learn</requirement>
+        <requirement type="package" version="0.7.0">imbalanced-learn</requirement>
+        <requirement type="package" version="1.2.5">pandas</requirement>
+        <requirement type="package" version="0.13.2">seaborn</requirement>
+        <requirement type="package" version="1.23.5">numpy</requirement>
+        <!-- additional pin (the datamanager failed because of this .. so I'm adding this just be sure) -->
+        <requirement type="package" version="2.7.0">pulp</requirement> <!-- needs to be pinned, because the old-ish snakemake used in vs2 struggles with https://github.com/snakemake/snakemake/issues/2607 -->
```

Comment:
Shouldn't some of this, at least, go upstream into the bioconda recipe to benefit more people?
For my personal taste, take a look at https://github.com/galaxyproject/tools-iuc/blob/main/tools/pangolin/pangolin.xml
There we're only pinning key dependencies that would clearly have a direct impact on reproducibility in Galaxy, but leave the rest to bioconda.

Reply:
They have just chosen a different way to deploy their software. See https://github.com/jiarong/VirSorter2/issues/205#issuecomment-2181966480. Guess its kind of the typical way snakemake deployments are made, i.e. one conda env for the pipeline and one for the actual tools that are executed. Basically they install https://github.com/jiarong/VirSorter2/blob/master/virsorter/envs/vs2.yaml

But you are right. I can at least remove some that already in bioconda recipe (click and python). Can we losen the pinning for some? Don't know if this works. But I'm pretty sure that we can not remove to many of them.